### PR TITLE
Fix extracting flutter file to tmp folder instead of current folder

### PR DESCRIPTION
**Pull request by marandaneto**
Fix https://github.com/getsentry/publish/issues/1828

> Error:  Error creating target instance for pub-dev!
> Error:  Executable "dart" not found. Is it installed?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -47,7 +47,7 @@ RUN apt-get -qq update \
   && chmod +x /usr/local/bin/SymbolCollector.Console
 
 RUN curl -fsSL https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_3.7.1-stable.tar.xz -o /tmp/flutter_linux_3.7.1-stable.tar.xz \
-  && tar xf /tmp/flutter_linux_3.7.1-stable.tar.xz \
+  && tar xf /tmp/flutter_linux_3.7.1-stable.tar.xz -C /tmp \
```
Comment (marandaneto) [inline Dockerfile:50]: `PATH` is set to `/tmp/flutter/bin`, so this should fix.

**Review by chadwhitacre — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (asottile-sentry) [inline Dockerfile:50]: it would be more natural to put this in `/opt/flutter` instead of `/tmp`

**Review by asottile-sentry — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (marandaneto) [reply Dockerfile:50]: @asottile-sentry I'd rather do it in a separate PR and unblock myself asap, this is just missing for now, other things use `tmp` as well.
is this fine?

File: Dockerfile [hunk 2]
```diff
@@ -46,9 +46,9 @@ RUN apt-get -qq update \
   && rm /tmp/sym-collector.zip \
   && chmod +x /usr/local/bin/SymbolCollector.Console
 
-RUN curl -fsSL https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_3.7.1-stable.tar.xz -o /tmp/flutter_linux_3.7.1-stable.tar.xz \
-  && tar xf /tmp/flutter_linux_3.7.1-stable.tar.xz \
-  && rm /tmp/flutter_linux_3.7.1-stable.tar.xz
+RUN curl -fsSL https://storage.googleapis.com/flutter_infra_release/releases/stable/linux/flutter_linux_3.7.1-stable.tar.xz -o /opt/flutter_linux_3.7.1-stable.tar.xz \
+  && tar xf /opt/flutter_linux_3.7.1-stable.tar.xz -C /opt \
+  && rm /opt/flutter_linux_3.7.1-stable.tar.xz
```
Comment (asottile-sentry) [inline Dockerfile:51]: it doesn't matter but the tgz should sit in `/tmp` but the unpack should be in `/opt`

**Review by asottile-sentry — APPROVED** [review]

**Pull request merged**